src/librustc_trait_selection/traits/error_reporting/suggestions.rs

```diff
@@ -760,13 +770,50 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
             #using-trait-objects-that-allow-for-values-of-different-types>";
         let has_dyn = snippet.split_whitespace().next().map_or(false, |s| s == "dyn");
         let trait_obj = if has_dyn { &snippet[4..] } else { &snippet[..] };
-        if all_returns_have_same_type {
+        if count == 0 {
+            // No return paths. Suggest `-> T`, `-> impl Trait`, and if `Trait` is object safe,
+            // `-> Box<dyn Trait>`.
+            err.note(
+                "currently nothing is being returned, depending on the final implementation \
+                 you could change the return type in different ways",
```

Comment:
```suggestion
                 change the return type in different ways",
```

Reply:
"nothing is being returned, depending on the final implementation, change the return type in different ways" feels a bit stilted.

Reply:
> nothing is being returned. Depending on the final implementation, change the return type in different ways

then?

Reply:
What about "depending on the final implementation, you have alternatives"? Or I can just disregard, remove the note and let users figure it out themselves that they can't apply all three suggestions simultaneously.

Reply:
Tweaked slightly:
> depending on the final implementation, here are some alternatives